### PR TITLE
Refactor test suite

### LINKED ISSUE #229: Split tests from index.js into separate files
Test cases in this libraries are all packed into one big `test/index.js` file. This makes it hard to work with the tests.

Tests should be broken to separate files analog to the module structure in the [src folder](https://github.com/ethereumjs/ethereumjs-util/tree/master/src) to make it easier to work on specific functionality and have an overview on functionality coverage.

Would be nice to have this for continued work on #228, no strict precondition though (of course).

### PR DESCRIPTION
**Pull request by ryanio**
This PR:
1. Splits single js test file into ts spec files (closes #229)
1. Tests src instead of dist
   1. normalizing effort based on https://github.com/ethereumjs/ethereumjs-vm/issues/561#issuecomment-577371591
1. Moves to github actions from travis
1. Runs karma in both headless firefox and chrome

### CODE REVIEW

File: src/constants.ts [hunk 1]
```diff
@@ -1,4 +1,5 @@
 import BN = require('bn.js')
+const Buffer = require('buffer').Buffer
```
Comment (ryanio) [inline src/constants.ts]: This was a workaround that was needed to fix the karma test runner.

File: karma.conf.js [hunk 2]
```diff
@@ -1,51 +1,28 @@
...
-        ]
-      ],
-      debug: true
+    plugins: ['karma-mocha', 'karma-typescript', 'karma-chrome-launcher', 'karma-firefox-launcher'],
+    karmaTypescriptConfig: {
+      bundlerOptions: {
+        entrypoints: /\.spec\.ts$/,
+        resolve: {
+          alias: {
+            'safe-buffer': 'node_modules/buffer/index.js',
```
Comment (ryanio) [inline karma.conf.js]: This was a workaround that was needed to fix the karma test runner.

**Comment by holgerd77** [comment]
Hi @ryanio, thanks for tackling this so quickly! 😄

Some initial notes: this PR is mixing up too much stuff, things should generally be addressed in a more atomic way for a PR to be better reviewable and also for - eventually necessary - later-on error/bug attribution. So these should rather be two (for the file split and the GitHub action transition) or eventually three (also for the Karma changes) PRs. Same a bit for the commits being done: changes are too big to pack this into a single commit, also for reasons of historical referenceability and again ease of review. The commit message along is also too generic to be meaningful enough. So there should be at least 4 separate commits attributed to the items identified above, eventually also some more and a bit more fine-grained. I e.g. discovered some implicit but significant changes (switch from Istanbul to nyc) hidden in the code base without any mentioning.

Won't make all of this a blocker for economic reasons, just as a note for future PRs. Think in this case this is justifiable to some extend since all changes are at least in the same scope of tests and at the end this can also be seen as some larger refactoring on this front.

Ok, so far. Apart from this all this is really a large improvement for the library, cool!

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    tags:
+      - '*'
+  pull_request:
+    types: [opened, reopened, synchronize]
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        node-version: [8.x, 10.x, 12.x, 13.x]
```
Comment (holgerd77) [inline .github/workflows/build.yml]: For the record: this PR removes support for Node 11 and adds support for Node 13.

File: karma.conf.js [hunk 4]
```diff
@@ -1,51 +1,23 @@
...
-      }
-    }
+    colors: true,
+    reporters: ['progress', 'karma-typescript'],
+    browsers: ['FirefoxHeadless', 'ChromeHeadless'],
+    singleRun: true,
+    concurrency: Infinity,
+    // Fail after timeout
+    browserDisconnectTimeout: 100000,
+    browserNoActivityTimeout: 100000,
```
Comment (holgerd77) [inline karma.conf.js]: Could you add some notes on the config you changed here together with some reasoning along?

File: package.json [hunk 5]
```diff
@@ -100,24 +98,19 @@
...
     "karma": "^4.0.0",
-    "karma-browserify": "^5.0.0",
     "karma-chrome-launcher": "^2.0.0",
-    "karma-detect-browsers": "2.2.6",
-    "karma-env-preprocessor": "^0.1.1",
     "karma-firefox-launcher": "^1.0.0",
     "karma-mocha": "^1.3.0",
+    "karma-typescript": "^4.1.1",
     "mocha": "^6.0.0",
+    "nyc": "^15.0.0",
```
Comment (holgerd77) [inline package.json]: For the record: switched from `istanbul` to `nyc`, dropped various `babel` and `karma` related dev dependencies.

File: test/account.spec.ts [hunk 6]
```diff
@@ -0,0 +1,530 @@
...
+    assert.equal(isValidAddress('0x52908400098527886E0F7030069857D2E4169EE7'), true)
+  })
+  it('should return false', function() {
+    assert.equal(isValidAddress('2f015c60e0be116b1f0cd534704db9c92118fb6a'), false)
+    assert.equal(isValidAddress('0x2f015c60e0be116b1f0cd534704db9c92118fb6'), false)
+    assert.equal(isValidAddress('0x2f015c60e0be116b1f0cd534704db9c92118fb6aa'), false)
+    assert.equal(isValidAddress('0X52908400098527886E0F7030069857D2E4169EE7'), false)
+    assert.equal(isValidAddress('x2f015c60e0be116b1f0cd534704db9c92118fb6a'), false)
+  })
+})
```
Comment (holgerd77) [inline test/account.spec.ts]: Ok (whole file).

File: test/object.spec.ts [hunk 7]
```diff
@@ -0,0 +1,134 @@
...
+
+  it('alias should work #2', function() {
+    var someOb: any = {}
+    var data = { blah: '0x1' }
+
+    defineProperties(someOb, fields, data)
+    assert.equal(someOb.blah.toString('hex'), '01')
+    assert.equal(someOb.aword.toString('hex'), '01')
+  })
+})
```
Comment (holgerd77) [inline test/object.spec.ts]: The stuff from `defineFields.js`, good.

File: test/signature.spec.ts [hunk 8]
```diff
@@ -0,0 +1,213 @@
...
+      '0x00000000000000000000000000000000000000004a1579cf389ef88b20a1abe90000000000000000000000000000000000000000326fa689f228040429e3ca661b',
+    )
+  })
+
+  it('should throw on invalid v value', function() {
+    assert.throws(function() {
+      toRpcSig(1, r, s)
+    })
+  })
+})
```
Comment (holgerd77) [inline test/signature.spec.ts]: 6 new files for 6 category source files, ok.

**Comment by holgerd77** [comment]
Also made the switch from the Travis to the GitHub action tests run being required and also included Coveralls to be now required (together with some reasonable thresholds) along the way.

Will now merge.

File: karma.conf.js (hunk 4 above)
Comment (ryanio) [reply karma.conf.js]: I grabbed some of the sensible defaults that were being used in ethereumjs-vm/karma.conf.js. I was going to bring over the comments indicating what each line does but decided not to for brevity, next time I will. Thanks!

**Pull request merged**